From 7622a0a4c6309f9cecaeef72569da687eb5b6b93 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Mon, 5 Apr 2021 01:13:29 +0200
Subject: [PATCH] pthread-cond: Fix wrong configure results when <pthread.h> is
 absent.

* m4/pthread-cond.m4 (gl_PTHREAD_COND): When <pthread.h> does not exist,
set HAVE_PTHREAD_COND_* to 0, not to 1.

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[yann.morin.1998@free.fr:
  - backport from upstream gnulib
  - drop upstream's Changelog hunk
  - reisntate Fabrice's SoB
]
Signed-off-by: Yann E. MORIN <yann.morin.1998@free.fr>

---
 m4/pthread-cond.m4 | 18 +++++++++---------
 2 files changed, 15 insertions(+), 9 deletions(-)

diff --git a/m4/pthread-cond.m4 b/m4/pthread-cond.m4
index 75e37c6138..8922a8a243 100644
--- a/m4/pthread-cond.m4
+++ b/m4/pthread-cond.m4
@@ -1,5 +1,5 @@
-# pthread-cond.m4 serial 1
-dnl Copyright (C) 2019-2020 Free Software Foundation, Inc.
+# pthread-cond.m4 serial 2
+dnl Copyright (C) 2019-2021 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -23,14 +23,14 @@ AC_DEFUN([gl_PTHREAD_COND],
     REPLACE_PTHREAD_COND_DESTROY=1
   else
     if test $HAVE_PTHREAD_H = 0; then
-      HAVE_PTHREAD_COND_INIT=1
-      HAVE_PTHREAD_CONDATTR_INIT=1
-      HAVE_PTHREAD_CONDATTR_DESTROY=1
-      HAVE_PTHREAD_COND_WAIT=1
-      HAVE_PTHREAD_COND_TIMEDWAIT=1
-      HAVE_PTHREAD_COND_SIGNAL=1
-      HAVE_PTHREAD_COND_BROADCAST=1
-      HAVE_PTHREAD_COND_DESTROY=1
+      HAVE_PTHREAD_COND_INIT=0
+      HAVE_PTHREAD_CONDATTR_INIT=0
+      HAVE_PTHREAD_CONDATTR_DESTROY=0
+      HAVE_PTHREAD_COND_WAIT=0
+      HAVE_PTHREAD_COND_TIMEDWAIT=0
+      HAVE_PTHREAD_COND_SIGNAL=0
+      HAVE_PTHREAD_COND_BROADCAST=0
+      HAVE_PTHREAD_COND_DESTROY=0
     fi
   fi
 ])
-- 
2.30.2

